### DISCUSSION TITLE
Is it possible to annotate Generator/Traversable/Iterable non-empty?

### DISCUSSION THREAD

**Discussion by mateusz-wu** [vimeo/psalm#9191] [Q&A]
I have a method returning `Generator<string>`. 
When passing it's result to a method requiring `non-empty-list/array` of course I get `"possibly undefined string"` 
Of course if it was about arrays/lists I'd return non-empty-array and problem would be solved, but not with Generators/Iterables.

Is it possible to annotate Generator/iterable non empty? 

Thanks a lot

**Comment by orklah** [comment]
It's not possible right now, I suggest just suppressing for now
